### DISCUSSION TITLE
Facing an Issue with fields under populate

### DISCUSSION THREAD

**Discussion by amitkTechno** [mikro-orm/nestjs#110] [Q&A]
I am facing a new issue which is related to [https://github.com/mikro-orm/mikro-orm/discussions/3012](https://github.com/mikro-orm/mikro-orm/discussions/3012)

Here I have 3 tables relationship
**1. User entity** 
```
@OneToMany(() => Shop, shop => shop.createdBy)
    shop = new Collection<Shop>(this);
@OneToMany(() => Product, product => product.owner)
    product = new Collection<Product>(this);
```
**2. Shop Entity**
```
@OneToMany(() => Product, product => product.shop)
    products = new Collection<Product>(this);
@ManyToOne()
    createdBy!: User;
```

**3. Products Entity**
```
@ManyToOne()
    shop!: Shop;
@ManyToOne()
    owner!: User;
```
`await this.shopRepository.findOne({ id: 1 },{ fields: ['*', 'createdBy.name', 'products.owner.email'], populate: ['createdBy', 'products.owner'] })`

**Result**
 ```
"id": 1,
"createdAt": "2023-03-14T06:02:33.000Z",
"updatedAt": "2023-03-14T06:02:33.000Z",
"name": "Shop1",
"description": "Shop1 Description",
"products": [
    {
        "id": 1,
        "shop": 1,
        "owner": {
            "id": 1,
            "name": "User One",
            "email": "userone@yopmail.com"
        }
    },
    {
        "id": 3,
        "shop": 1,
        "owner": {
            "id": 2,
            "email": "usertwo@yopmail.com"
        }
    }
],
"createdBy": {
    "id": 1,
    "name": "User One",
    "email": "userone@yopmail.com"
}
```
**Expected behavior**
The result should not contain owner name inside products owner object (**which owner id is 1**).

**Versions**

- NodeJs - 16.16.0
- Mikro-orm - 5.5.3
- Database - Postgres | MySQL

Thanks @B4nan

**Comment by amitkTechno** [comment]
Please help me regarding this @B4nan

**Reply by B4nan** [reply]
You might have to wait for v6, fixing this will be a bit breaking so won't do that in 5.x release.

Subscribe here https://github.com/mikro-orm/mikro-orm/issues/4138

And for now, use can use explicit serialization

https://mikro-orm.io/docs/serializing#explicit-serialization
